ui/js/graphs.js

```diff
@@ -769,6 +786,73 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
     $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.selectedTestsToAdd = []; // tests in the "to add" test list that have been selected by the user
 
+    var relateSeries = options.relatedSeries;
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    var addRelatePlatform = function(relateSeries) {
+        PhSeries.getAllSeries(relateSeries.projectName,
+            $scope.timeRange, optionCollectionMap).then(
+            function(seriesData) {
+                var platformList = seriesData.platformList;
+                platformList.forEach(function(platform) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                        { platform: platform }), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.name});
+                    // if found something different from the series we already have,
+                    // then we push it into the testsToAdd list.
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+            }
+         );
+    };
+    // if we choose to add all related branches, we should
+    // to find out all the branches of specific platform. Then query
+    // the test data one by one.
+    var addRelateBranches = function(options, relateSeries) {
+        var branchList = [];
+        theBranchedAboutPerform.forEach(function(branch) {
+            if (branch !== options.relatedSeries.projectName) {
+                branchList.push(_.findWhere($scope.projects,
+                    {name: branch}));
+            }
+        });
+        branchList.forEach(function(project) {
+            PhSeries.getAllSeries(project.name,
+                $scope.timeRange, optionCollectionMap).then(
+                function(seriesData) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                        {platform: relateSeries.platform}), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.name});
+                    $scope.testsToAdd.push(_.clone(temp));
+                }
+            );
+        });
+    };
+    $scope.addRelated = function() {
+        if(options.option === "addRelatedPlatforms") {
+            addRelatePlatform(relateSeries);
+        } else if(options.option === "addRelatedBranchs") {
+            addRelateBranches(options, relateSeries);
+        }
```

Comment:
These functions could take a long time, so they should work asynchronously with some kind of progress indicator in the foreground while this is happening. You can see what we did here with the comparison view for an example:

https://github.com/mozilla/treeherder/blob/master/ui/partials/perf/comparectrl.html#L3

(we should probably use something other than a dancing cat here, we don't want to overuse that joke) :)